pkg/services/dashboards/database/database.go

```diff
@@ -1055,3 +1058,123 @@ func readQuotaConfig(cfg *setting.Cfg) (*quota.Map, error) {
 	limits.Set(orgQuotaTag, cfg.Quota.Org.Dashboard)
 	return limits, nil
 }
+
+func (d *dashboardStore) SaveK8sDashboard(ctx context.Context, orgID int64, uid string, meta metav1.ObjectMeta, data *simplejson.Json) (*dashboards.Dashboard, error) {
+	var result *dashboards.Dashboard
+	var err error
+
+	data.Del("id") // ignore any internal id
+	anno := crd.CommonAnnotations{}
+	anno.Read(meta.Annotations)
+
+	if anno.CreatedAt < 1 {
+		anno.CreatedAt = time.Now().UnixMilli()
+	}
+	if anno.UpdatedAt < 1 {
+		anno.UpdatedAt = time.Now().UnixMilli()
+	}
+	if anno.OrgID < 1 {
+		anno.OrgID = 1 // TODO, set from namespace?
+	}
+
+	dash := &dashboards.Dashboard{
+		UID:       uid,
+		OrgID:     anno.OrgID,
+		Data:      data,
+		Created:   time.UnixMilli(anno.CreatedAt),
+		CreatedBy: anno.CreatedBy,
+		Updated:   time.UnixMilli(anno.UpdatedAt),
+		UpdatedBy: anno.UpdatedBy,
+
+		// Plugin provisioning
+		PluginID: anno.PluginID,
+	}
+
+	err = d.store.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
+		result, err = saveK8sDashboard(sess, dash, anno.FolderUID, d.emitEntityEvent())
+		if err == nil && anno.OriginKey != "" {
+			p := &dashboards.DashboardProvisioning{
+				Name:        anno.OriginName,
+				ExternalID:  anno.OriginPath,
+				CheckSum:    anno.OriginKey,
+				Updated:     anno.OriginTime,
+				DashboardID: result.ID,
+			}
+			return saveProvisionedData(sess, p, result)
+		}
+		return err
+	})
+	return result, err
+}
+
+func saveK8sDashboard(sess *db.Session, dash *dashboards.Dashboard, folderUID string, emitEntityEvent bool) (*dashboards.Dashboard, error) {
+	var existing dashboards.Dashboard
+	dashWithUIDExists, err := sess.Where("uid=? AND org_id=?", dash.UID, dash.OrgID).Get(&existing)
+	if err != nil {
+		return nil, err
+	}
+
+	var affectedRows int64
+	if dashWithUIDExists {
+		dash.ID = existing.ID
+		dash.SetVersion(existing.Version + 1)
+		affectedRows, err = sess.MustCols("folder_id").ID(dash.ID).Update(dash)
```

Comment:
I don't understand the "folder_id" constraint -- I am sure something is off here if the folder id changes from one request to the next

Reply:
I'll leave it... but we don't even use the results here so 🤷